.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,266 @@
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      version: ${{ steps.supported-versions-matrix.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-versions-matrix
+        uses: WyriHaximus/github-action-composer-php-versions-in-range@v1
+
+  supported-checks-matrix:
+    name: Determine which checks to run
+    runs-on: ubuntu-latest
+    needs: [config]
+    outputs:
+      check: ${{ steps.supported-checks-matrix.outputs.check }}
+    env:
+      CONFIG_JSON: ${{ needs.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-checks-matrix
+        run: |
+          # This grabs the default known checks from the laminas.dev API.
+          #
+          # If the .github/workflows/continuous-integration-config.json file is
+          # present, and defines a "checks" array, that value will be
+          # used as the list of known checks.
+          #
+          # This list is then compared to the list of scripts in composer.json,
+          # and the intersection of them is provided as the output.
+          #
+          # If you want to use fewer or more checks than the defaults (e.g., to run
+          # benchmarks provided in the current repository, or NOT run static
+          # analysis, even if the scripts are present), create and/or update the
+          # continuous-integration-config.json file. The value should include all
+          # checks which should be run.
+          JQ_SCRIPT_TEMPLATE='def tools:TOOLS ; def intersect(x;y): (x + y | sort) as $sorted | reduce range(1; $sorted|length) as $i ([]; if $sorted[$i] == $sorted[$i-1] then . + [$sorted[$i]] else . end) ; .scripts | keys | intersect(.;tools)'
+          DEFAULT_CHECKS=$(curl -s https://laminas.dev/api/ci/default-checks | jq -c '.checks')
+          CONFIG_CHECKS=$(echo "${CONFIG_JSON}" |  jq -c 'def isarray(x): if x|type=="array" then x else empty end; isarray(.checks)')
+          ALL_CHECKS=${CONFIG_CHECKS:-$DEFAULT_CHECKS}
+          JQ_SCRIPT=${JQ_SCRIPT_TEMPLATE/TOOLS/$ALL_CHECKS}
+          CHECKS=$(jq -c "$JQ_SCRIPT" < composer.json)
+          echo "Discovered checks: ${CHECKS}"
+          echo "::set-output name=check::${CHECKS}"
+
+  stable-php:
+    name: Determine Stable PHP Version for Non-Unit Test Checks
+    runs-on: ubuntu-latest
+    needs: [config, supported-versions-matrix]
+    outputs:
+      version: ${{ steps.stable-php.outputs.version }}
+    env:
+      CONFIG_JSON: ${{ needs.config.outputs.json }}
+    steps:
+      - id: stable-php
+        run: |
+          # This grabs the default stable PHP version for non-unit-test checks from the laminas.dev API.
+          # If the .github/workflows/continuous-integration-config.json file is
+          # present, and defines a string "stable-php" element, that value will
+          # be used.
+          DEFAULT_VERSION=$(curl -s https://laminas.dev/api/ci/stable-checks-version | jq -r '.version')
```

Comment:
Similar to above comments, the information should probably be pulled off the current github action that contains information about the latest stable PHP version (or queries it on php.net as fallback, but at least it's a local script first)